src/traces/sankey/attributes.js

```diff
@@ -0,0 +1,149 @@
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+
+module.exports = {
+    hoverinfo: extendFlat({}, plotAttrs.hoverinfo, {
+        flags: ['label', 'text', 'value', 'percent', 'name']
+    }),
+    domain: {
+        x: {
+            valType: 'info_array',
+            role: 'info',
+            items: [
+                {valType: 'number', min: 0, max: 1},
+                {valType: 'number', min: 0, max: 1}
+            ],
+            dflt: [0, 1],
+            description: [
+                'Sets the horizontal domain of this `sankey` trace',
+                '(in plot fraction).'
+            ].join(' ')
+        },
+        y: {
+            valType: 'info_array',
+            role: 'info',
+            items: [
+                {valType: 'number', min: 0, max: 1},
+                {valType: 'number', min: 0, max: 1}
+            ],
+            dflt: [0, 1],
+            description: [
+                'Sets the vertical domain of this `sankey` trace',
+                '(in plot fraction).'
+            ].join(' ')
+        }
+    },
+
+    orientation: {
+        valType: 'enumerated',
+        values: ['v', 'h'],
+        dflt: 'h',
+        role: 'style',
+        description: 'Sets the orientation of the Sankey diagram.'
+    },
+
+    nodepad: {
+        valType: 'number',
+        min: 0,
+        dflt: 20,
+        role: 'style',
+        description: 'Sets the padding (in px) between the `nodes`.'
+    },
+
+    nodethickness: {
+        valType: 'number',
+        min: 1,
+        dflt: 20,
+        role: 'style',
+        description: 'Sets the thickness of the `nodes`.'
+    },
+
+    valueformat: {
+        valType: 'string',
+        dflt: '.3s',
+        role: 'style',
+        description: [
+            'Sets the value formatting rule using d3 formatting mini-language',
+            'which is similar to those of Python. See',
+            'https://github.com/d3/d3-format/blob/master/README.md#locale_format'
+        ].join(' ')
+    },
+
+    valueunit: {
+        valType: 'string',
+        dflt: '',
+        role: 'style',
+        description: [
+            'Adds a unit to follow the value in the hover tooltip. Add a space if a separation',
+            'is necessary from the value.'
+        ].join(' ')
+    },
+
+    // followmouse for UX testing
+    followmouse: {
+        valType: 'boolean',
+        dflt: true,
+        role: 'style',
+        description: 'Sets whether the hover tooltip follows the mouse.'
+    },
+
+    textfont: fontAttrs,
+
+    nodes: {
+        _isLinkedToArray: 'node',
+        label: {
+            valType: 'string',
+            dflt: '',
+            role: 'info',
+            description: 'The shown name of the node.'
+        },
+        visible: shapeAttrs.visible,
+        color: extendFlat({}, shapeAttrs.fillcolor, {
+            dflt: 'rgb(0,255,0,0.5)'
+        }),
+        description: 'The nodes of the Sankey plot.'
+    },
+
+    links: {
```

Comment:
Thanks, tried to apply things learnt off the bat with this new chart; still room for improv!